[DOCS] Create rule updates 8.4 (#2339)
elastic/security-docs: docs/detections/rules-ui-create.asciidoc

{ref}/eql.html[Event Query Language (EQL)] query.
* <<create-indicator-rule, *Indicator match*>>: Creates an alert when {elastic-sec} index field values match field values defined in the specified indicator index patterns. For example, you can create an indicator index for IP addresses and use this index to create an alert whenever an event's `destination.ip` equals a value in the index. Indicator index field mappings should be {ecs-ref}[ECS-compliant]. For information on creating {es} indices and field types, see
{ref}/getting-started-index.html[Index some documents],
{ref}/indices-create-index.html[Create index API], and
{ref}/mapping-types.html[Field data types]. If you have indicators in a standard file format, such as CSV or JSON, you can also use the Machine Learning Data Visualizer to import your indicators into an indicator index. See {ml-docs}/ml-getting-started.html#sample-data-visualizer[Explore the data in {kib}] and use the *Import Data* option to import your indicators.

TIP: You can also use value lists as the indicator match index. See <<indicator-value-lists>> at the end of this topic for more information.

> * <<create-new-terms-rule, *New terms*>>: Generates an alert for each new term it detects in source documents.

Reviewer: Can we add a bit to define "new term" here and how the History Window Size works? E.g. for a rule with an interval of 5 minutes, 0 additional lookback, and a history window size of 7 days, a term will be "new" if the only time it appears within the last 7 days is also within the last 5 minutes.

Reviewer 2: +1 for a concise definition of "new term." Maybe for this first reference, generalize about time intervals and history windows, and provide those details in the "Create a new terms rule" section below? This sort of gets at that, though it could probably still use more explanation of what "new" means:
```suggestion
* <<create-new-terms-rule, *New terms*>>: Generates an alert for each new term detected in source documents within a specified time period.
```

==== Create a custom query rule
. To create a rule based on a KQL or Lucene query, select *Custom query*,
then:
.. Define which {es} indices or data view the rule searches for alerts.
.. Use the filter and query fields to create the criteria used for detecting
alerts.
+

> NOTE: You can use {kib} saved queries (image:images/saved-query-menu.png[Saved query menu,17,17]) and queries from saved Timelines (`Import query from saved Timeline`) as rule conditions.

Reviewer 2: I think the Timelines link should be bold here, since it's a label/link the user will click. Monospace is typically more for values that the user enters, or code.
```suggestion
NOTE: You can use {kib} saved queries (image:images/saved-query-menu.png[Saved query menu,17,17]) and queries from saved Timelines (*Import query from saved Timeline*) as rule conditions.
```

+
. Continue with <<preview-rules, previewing the rule>> (optional) or click *Continue* to <<rule-ui-basic-params, configure basic rule settings>>.

[discrete]
[[create-new-terms-rule]]

> ==== Create a new terms rule

Reviewer 3: ```suggestion
==== Create a New Terms rule
```
proper noun?

Reviewer 2: My vote would be lowercase here, following the general usage of the other rule type names.

. To create a rule that searches for each new term detected in source documents, select *New Terms*, then:
.. Specify what data to search by entering individual {es} index patterns or selecting an existing data view.

> .. Use the filter and query fields to create the criteria used for detecting

Reviewer 3: ```suggestion
.. Use the filter and query fields to create the criteria used for alerting.
```

> alerts.

Reviewer 3: ```suggestion
```

+

> NOTE: NOTE: You can use {kib} saved queries (image:images/saved-query-menu.png[Saved query menu,17,17]) and queries from saved Timelines (`Import query from saved Timeline`) as rule conditions.

Reviewer 3: ```suggestion
NOTE: You can use {kib} saved queries (image:images/saved-query-menu.png[Saved query menu,17,17]) and queries from saved Timelines (`Import query from saved Timeline`) as rule conditions.
```

Reviewer 2: I think the Timelines link should be bold here, since it's a label/link the user will click. Monospace is typically more for values that the user enters or code.
```suggestion
NOTE: You can use {kib} saved queries (image:images/saved-query-menu.png[Saved query menu,17,17]) and queries from saved Timelines (*Import query from saved Timeline*) as rule conditions.
```

+
.. Use the `Fields` menu to select a field to check for new terms.

> .. Use the `History Window Size` to specify the time range to search in minutes, hours, or days to determine if a term is new.

Reviewer: It may be worth adding a bit more detail about how the history window size and rule interval work together. The history window size must be larger than the rule interval + additional lookback, because the rule is looking for terms where the only time(s) the term appears within the history window are also within the rule interval + additional lookback.

> .. Use the `Fields` menu to select a field to check for new terms.
> .. Use the `History Window Size` to specify the time range to search in minutes, hours, or days to determine if a term is new.

Reviewer 2: Same thing with bold here -- these are UI elements, not entered values or code examples:
```suggestion
.. Use the *Fields* menu to select a field to check for new terms.
.. Use the *History Window Size* to specify the time range to search in minutes, hours, or days to determine if a term is new.
```

> ..Continue with <<preview-rules, previewing the rule>> (optional) or click *Continue* to <<rule-ui-basic-params, configure basic rule settings>>.

Reviewer 4: ```suggestion
.. Continue with <<preview-rules, previewing the rule>> (optional) or click *Continue* to <<rule-ui-basic-params, configure basic rule settings>>.
```

+
NOTE: For threshold rules, not all source event values can be used for overrides; only the fields that were aggregated over (the `Group by` fields) will contain data.
.. *Timestamp override* (optional): Select a source event timestamp field. When selected, the rule's query uses the selected field, instead of the default `@timestamp` field, to search for alerts. This can help reduce missing alerts due to network or server outages. Specifically, if your ingest pipeline adds a timestamp when events are sent to {es}, this avoids missing alerts due to ingestion delays.

> However, if you know your data source has an inaccurate `@timestamp` value, it is recommended you check the *Do not use @timestamp as a fallback timestamp field* option to ignore the `@timestamp` field entirely.

Reviewer 4: ```suggestion
However, if you know your data source has an inaccurate `@timestamp` value, it is recommended you select the *Do not use @timestamp as a fallback timestamp field* option to ignore the `@timestamp` field entirely.
```

+
TIP: The {filebeat-ref}/filebeat-module-microsoft.html[Microsoft] and

> {filebeat-ref}/filebeat-module-google_workspace.html[Google Workspace] Filebeat modules have an `event.ingested` timestamp field that can be used instead of the default `@timestamp` field.

Reviewer 4: ```suggestion
{filebeat-ref}/filebeat-module-google_workspace.html[Google Workspace] {filebeat} modules have an `event.ingested` timestamp field that can be used instead of the default `@timestamp` field.
```
